core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js

```diff
@@ -138,6 +148,49 @@ oppia.factory('SimpleEditorManagerService', [
         data.questionList.addQuestion(QuestionObjectFactory.create(
           lastStateName, stateData.interaction, ''));
       },
+      deleteQuestion: function(question){
+        // - Change destination of answer groups that point to it.
+        // - Move content stored in present state to next state if it exists.
+        // - Delete the state.
+        var stateName = question.getStateName();
+        var state = SimpleEditorShimService.getState(stateName);
+        // If it's the last question in the list, make it EndExploration.
+        if(state.interaction.answerGroups.length == 0){
+          makeStateEndExploration(stateName);
+          data.questionList.removeQuestion(question);
+          return;
+        }
+        var nextStateName = state.interaction.answerGroups[0].outcome.dest;
+        var allStateNames = SimpleEditorShimService.getAllStateNames();
+        // Change init state name, if init_state is being deleted.
+        if(SimpleEditorShimService.getInitStateName() == stateName){
+          explorationInitStateNameService.displayed = nextStateName;
+          explorationInitStateNameService.saveDisplayedValue(nextStateName);
+        }
+
+        for(var i = 0; i < allStateNames.length;i++){
+          var currentState = SimpleEditorShimService.getState(allStateNames[i]);
+          var newAnswerGroups = currentState.interaction.answerGroups;
+          var hasAnswerGroupsChanged = false;
+          for(var j = 0; j < currentState.interaction.answerGroups.length;j++){
```

Comment:
It might be clearer to use a forEach loop here, since you don't really need the index:

    currentState.interaction.answerGroups.forEach(function(answerGroup) {
      ...
    });